Patterns/English/English-DateTime.yaml

```diff
@@ -553,10 +553,14 @@ HolidayList3: !simpleRegex
 # Islamic holidays
 HolidayList4: !simpleRegex
   def: (?<holiday>ramad(h)?an|ram(a)?zan|ramathan|eid al(-|\s+)adha|eid al(-|\s+)azha|eidul(-|\s+)azha|feast of the sacrifice|(islamic|arabic|hijri) new year|eid al(-|\s+)fitr|festival of breaking the fast)
+# including weekend with holiday names
+HolidayWeekendRegex: !nestedRegex
+  def: (?<holidayWeekend>(({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+weekend)(\s+((of\s+)?({YearRegex}|{RelativeRegex}\s+year))))|(({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+((of\s+)?({YearRegex}|{RelativeRegex}\s+year)))?(\s+weekend))|(((the\s+)?weekend\s+of\s+)({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+((of\s+)?({YearRegex}|{RelativeRegex}\s+year)))?))
+  references: [ HolidayList1, HolidayList2, HolidayList3, HolidayList4, YearRegex, RelativeRegex, StrictRelativeRegex ]
 # Combined holiday regex
 HolidayRegex: !nestedRegex
-  def: \b(({StrictRelativeRegex}\s+({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4}))|(({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+(of\s+)?({YearRegex}|{RelativeRegex}\s+year))?))\b
-  references: [ HolidayList1, HolidayList2, HolidayList3, HolidayList4, YearRegex, RelativeRegex, StrictRelativeRegex ]
+  def: \b({HolidayWeekendRegex}|({StrictRelativeRegex}\s+({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4}))|(({HolidayList1}|{HolidayList2}|{HolidayList3}|{HolidayList4})(\s+(of\s+)?({YearRegex}|{RelativeRegex}\s+year))?))\b
```

Comment:
This still duplicates all holidays list both inside HolidayWeekendRegex and HolidayRegex. can't the "weekend" piece itself be the named group? No need to capture the holiday names along with it.

Reply:
I've merged the regexes. There are three patterns for holiday weekends:
-Halloween weekend (year)?
-Halloween (year)? weekend
-Weekend of Halloween (year)?

Instead of repeating the holiday list for each pattern, I've merged the first two into HolidayRegex, and needed to repeat the holiday list once for the third pattern. I can remove this repetition but this would capture incorrect input patterns like:

"weekend of halloween weekend 2021" or "weekend of halloween 2021 weekend"

For that reason, I will keep the repetition for the third pattern (which starts with a weekend prefix, so most strings would be rejected quickly). If you would prefer removing the extra repetition, I can do hotfix on this early morning tomorrow before we merge.